frontend/views/components/Modal/Modal.vue

```diff
@@ -35,31 +38,38 @@ export default {
   },
   watch: {
     '$route' (to, from) {
-      if (to.query.modal) {
-        // We reset the modals with no animation for simplicity
-        if (to.query.modal !== this.content) this.content = to.query.modal
-        const subcontent = to.query.subcontent
-        if (subcontent !== this.activeSubcontent()) {
-          // Try to find the new subcontent in the list of subcontent
-          const i = this.subcontent.indexOf(subcontent)
-          if (i) {
-            this.subcontent = this.subcontent.splice(0, i)
-          } else this.subcontent = subcontent
+      if (this.isUrlChange) {
+        console.log('URLCHANGE')
+        if (to.query.modal) {
+          // We reset the modals with no animation for simplicity
+          if (to.query.modal !== this.content) this.content = to.query.modal
+          const subcontent = to.query.subcontent
+          if (subcontent !== this.activeSubcontent()) {
+            // Try to find the new subcontent in the list of subcontent
+            const i = this.subcontent.indexOf(subcontent)
+            if (i) {
+              this.subcontent = this.subcontent.splice(0, i)
```

Comment:
1. This should check `if (i !== -1)`, not `if (i)`
2. This usage of splice is extremely confusing and probably wrong. What are you trying to do here? Here is what it's doing:
  1. First it modifies `this.subcontent` by removing, starting from the 0th index, `i` elements. It is unclear why this is happening. If `i` is `0`, it removes nothing. If `i` is `1`, it removes the first element. It doesn't make sense why it's doing this.
  2. Then, the modified `this.subcontent` is again modified by replacing it with the return value from the call to `.splice`. If `i` is `0`, it becomes an empty array (`[]`), because nothing was removed. If `i` is `1`, then it becomes an array containing the first element. Again, it isn't clear to me what the logic here is trying to do or why

So this code three major problems:

1. It has a bug because `i` could be `-1`, and it doesn't check for that
2. It is very confusing code, and it's not at all clear what it's doing or why
3. It probably doesn't work (i.e. it probably doesn't support handling 3 modals, at most it handles 2 modals), so either it should be fixed to correctly handle an arbitrary amount of modals, or it should be modified to correctly handle only two modals, in which case `subcontent` should be a direct reference to the second modal, instead of an array of modal names (see discussion in #685)